### DISCUSSION TITLE
Using Postal for self hosted Discourse for SMTP

### DISCUSSION THREAD

**Discussion by robtec** [postalserver/postal#3037] [Help with using Postal]
I'm using postal (https://docs.postalserver.io/) and have tls enabled on port 25, for SMTP. I am trying to point my Discourse server to it but I get the following error in Discourses logs

```
Jobs::HandledExceptionWrapper: Wrapped OpenSSL::SSL::SSLError: SSL_connect SYSCALL returned=5 errno=0 peeraddr=x.x.x.x:25 state=SSLv3/TLS write client hello
```
My Discourse config would be

```
DISCOURSE_SMTP_ADDRESS: <address>
DISCOURSE_SMTP_PORT: 25
DISCOURSE_SMTP_USER_NAME: <username>
DISCOURSE_SMTP_PASSWORD: "<password>"
DISCOURSE_SMTP_ENABLE_START_TLS: true           # (optional, default true)
DISCOURSE_SMTP_DOMAIN: <domain>
DISCOURSE_NOTIFICATION_EMAIL: <no-reply>
DISCOURSE_SMTP_AUTHENTICATION: "plain"
DISCOURSE_SMTP_OPENSSL_VERIFY_MODE: none
```

Is this an issue with `STARTTLS`?

**Comment by willpower232** [comment]
Sounds like it, did you add a certificate in your config file?

**Reply by robtec** [reply]
I have a cert and key in the config folder, I'm guessing the cert needs to be one the maps to the domain I'm running postal on? for example mail.domain.com

A self signed one won't work right?

**Reply by willpower232** [reply]
You can confirm what your Postal is currently doing with SSL like this https://docs.postalserver.io/other/debugging#smtp-ssl

I think it depends on how the client is set up, its possible that Discourse is still verifying the certificate on some level or perhaps there is a mismatch of supported SSL settings. Usually, its relatively easy to re-use the certificate from the web UI as the SMTP certificate but that depends on how that works.
